odoo/modules/registry.py

```diff
@@ -244,7 +247,10 @@ def setup_models(self, cr):
             for model in env.values():
                 model._unregister_hook()
 
+        # clear cache to ensure consistency, but do not signal it
         self.clear_caches()
+        self.cache_invalidated = False
```

Comment:
```suggestion
        # clear cache to ensure consistency, but do not signal it
        self.__cache.clear()
```

Reply:
I initially did not do it because `clear_caches()` still looks like a semi-official API that some models may override, and that meant bypassing it. But it used to be done implicitly by `lazy_property.reset_all()` so I guess it was never guaranteed anyway.

Besides, this mechanism has been outdated for years, now that cache invalidation is global, so it looks more like a useless loop than anything.

We still have the same thing in `check_signaling()`, which I'm touching in the second commit. This one has been calling `clear_caches()` consistently for a while, so I'm not too fond of dropping the indirection in stable: https://github.com/odoo/odoo/blob/master/odoo/modules/registry.py#L599-L600